conans/test/unittests/client/tools/scm/test_git.py

```diff
@@ -74,6 +89,42 @@ def test_clone_git(self):
         git.clone(path)
         self.assertTrue(os.path.exists(os.path.join(tmp, "myfile")))
 
+    @parameterized.expand([(None,),  # default
+                           ("develop",),  # branch name
+                           ("1.0",),  # tag name
+                           ("HEAD",),  # expression
+                           ])
+    def test_clone_git_shallow(self, element):
+        path, _ = create_local_git_repo({"myfile": "contents"}, commits=3, tags=["1.0"], branch="develop")
+        tmp = temp_folder()
+        git = Git(tmp)
+        git.clone("file://" + path, branch=element, shallow=True)  # --depth is ignored in local clones
+        with self.assertRaises(subprocess.CalledProcessError):
+            git.checkout(element="HEAD~1")
+        self.assertTrue(os.path.exists(os.path.join(tmp, "myfile")))
```

Comment:
I would like to check that the _shallow_ has worked: `git rev-list --all --count` should be `1`.

And also, is the output for all the parameterized use cases returning the same revision? Can we check `git.get_revision()` against the revision returned by the `create_local_git_repo`?

Reply:
okay, adding that

Reply:
added additional validation